### PR TITLE
test: major Cypress upgrade (v9.5.0 -> v12.2.0)

### PR DESCRIPTION
**Pull request by ctschacher**
## Description

We had a quite old Cypress version in use (v9.5). Interesting new developments and features were added to Cypress, to take advantage of these we decided to upgrade to the latest version of Cypress (v12.2.0).

This upgrade required a lot of changes to the Cypress configuration, just to mention a few:
- Cypress config files are no longer in json format but typescript files
- Cypress now needs some extra start parameters when invokes from CLI
- extra module 'cypress-extends' was removed as the config can now be extended using common imports
- etc

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/test-cypress-migration-to-v12-run-e2e/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-zqtxwzuxyh.chromatic.com)

### CODE REVIEW

File: gravitee-apim-e2e/ui-test/integration/apim/ui/gateways/ui-gateway-information.spec.ts [hunk 1]
```diff
@@ -64,16 +64,16 @@ describe('Get Gateway instance information as admin', () => {
...
     cy.contains('Information');
     cy.contains('Plugins');
     cy.contains('System properties');
   });
 
   it('should display monitoring information of selected gateway', () => {
-    cy.getByDataTestId('instances_instances-box').its(0).trigger('click');
-    cy.get('[data-testid="management.instances.detail.monitoring"]').trigger('click');
+    cy.getByDataTestId('instances_instances-box').first().find('a').trigger('click');
+    cy.getByDataTestId('management.instances.detail.monitoring').trigger('click');
```
Comment (gaetanmaisse) [inline gravitee-apim-e2e/ui-test/integration/apim/ui/gateways/ui-gateway-information.spec.ts]: It looks like this test is failing after the update 🤔

Comment (ctschacher) [reply gravitee-apim-e2e/ui-test/integration/apim/ui/gateways/ui-gateway-information.spec.ts]: @gaetanmaisse Yes right, one test is failing but I think it was already failing before and I have the impression that it's actually related to a code change in the UI and for this I'd like some help. I will get in contact in Florent in the afternoon to talk about it. If you want and your time allows it, we could also check on that together. Otherwise, as mentioned, I hope to reach Florian later on.

File: gravitee-apim-e2e/ui-test/support/commands.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/*
```
Comment (loriepisicchio) [inline gravitee-apim-e2e/ui-test/support/commands.ts]: do we need to keep this ?

File: gravitee-apim-e2e/ui-test/support/e2e.ts [hunk 3]
```diff
@@ -21,4 +21,4 @@ import './common/api.commands';
 
 // Alternatively you can use CommonJS syntax:
 // require('./commands')
-require('cypress-terminal-report/src/installLogsCollector')({ enableExtendedCollector: Cypress.env('printLogsToConsole') === 'always' });
+// require('cypress-terminal-report/src/installLogsCollector')({ enableExtendedCollector: Cypress.env('printLogsToConsole') === 'always' });
```
Comment (loriepisicchio) [inline gravitee-apim-e2e/ui-test/support/e2e.ts]: 🗑️  ?

File: gravitee-apim-e2e/ui-test/support/commands.ts (hunk 2 above)
Comment (ctschacher) [reply gravitee-apim-e2e/ui-test/support/commands.ts]: Ok, I'll remove these comments and put references to the actual command definitions there (had them in e2e.ts file before) as this the place where they supposed to be.

File: gravitee-apim-e2e/ui-test/support/e2e.ts (hunk 3 above)
Comment (ctschacher) [reply gravitee-apim-e2e/ui-test/support/e2e.ts]: Alright, I'll remove that line and will also de-install the cypress-terminal-report module as it created more problems that benefits.

File: gravitee-apim-e2e/package.json [hunk 4]
```diff
@@ -9,10 +9,9 @@
         "build": "tsc --removeComments --outDir dist --target es5",
         "bulk": "jest -c jest.config.bulk.js -e --forceExit --logHeapUsage --useStderr --runInBand --detectOpenHandles",
         "test:ui": "sh ./scripts/run-docker-compose.sh ui-test mongo",
-        "test:ui:dev": "cypress open --config-file ui-test/integration/apim/ui/cypress-config.json",
-        "test:ui:bulk": "cypress run --config-file ui-test/integration/apim/bulk/cypress-config.json",
```
Comment (phiz71) [inline gravitee-apim-e2e/package.json:13]: why did you delete this line ?

**Review by phiz71 — COMMENTED** [review]
Just a question regarding `test:ui:bulk` command you deleted

File: gravitee-apim-e2e/package.json (hunk 4 above)
Comment (ctschacher) [reply gravitee-apim-e2e/package.json:13]: Ok, well spotted. Not sure how this happened, it was not done intentionally. I'll bring it back.

Comment (ctschacher) [reply gravitee-apim-e2e/package.json:13]: A little update on this:
It seems that the bulk file was once migrated and is now to be executed by Jest. In fact, there's a npm run script for bulk in line 10 (as seen in your excerpt of the package.json):
`"bulk": "jest -c jest.config.bulk.js -e --forceExit --logHeapUsage --useStderr --runInBand --detectOpenHandles",`

Which means that the `test:ui:bulk` entry is actually obsolete and I'll leave it as it is now.

**Review by phiz71 — APPROVED** [review]
